sympy/series/limits.py

```diff
@@ -319,8 +319,10 @@ def remove_abs(expr):
 
         try:
             if str(dir) == '+-':
-                r = gruntz(e, z, z0, '+')
-                l = gruntz(e, z, z0, '-')
+                r = limit(e, z, z0, '+')
```

Comment:
`gruntz` was put in a try-except block because it may raise. Is it necessary for `limit` as well?

Reply:
No it is not since `limit()` already has those checks in place. `limit()` for each individual direction will automatically call `
`gruntz()`.

Reply:
I think the try-except block is still needed since the block stil contains one `gruntz` statement for dir `+` or `-`.

That `gruntz` statement can't be removed so that block can't be removed.

Reply:
> I think the try-except block is still needed since the block stil contains one `gruntz` statement for dir `+` or `-`.


That try-except block is still there. I think @jksuom is referring to the one I added 
```
     try:
          r = gruntz(e, z, z0, '+')
      except (PoleError, ValueError):
          r = limit(e, z, z0, '+')
      try:
          l = gruntz(e, z, z0, '-')
      except (PoleError, ValueError):
          l = limit(e, z, z0, '-')
```

Reply:
I would move the `+-` case with the two `limit` calls in front of the try-except block as they are not expected to raise. Only `gruntz` would remain in the block.

Reply:
@jksuom  That is indeed the case now. This is how the [try-except block](https://github.com/sayandip18/sympy/blob/b0da3b8a4cc25e0d5e70b6162c64a8974ce5d202/sympy/series/limits.py#L320) looks right now.
```
       try:
            if str(dir) == '+-':
                r = limit(e, z, z0, '+')
                l = limit(e, z, z0, '-')
                if isinstance(r, Limit) or isinstance(l, Limit):
                    return Limit(e, z, z0, '+-')
                if l != r:
                    raise ValueError("The limit does not exist since "
                            "left hand limit = %s and right hand limit = %s"
                            % (l, r))
            else:
                r = gruntz(e, z, z0, dir)
            if r is S.NaN or l is S.NaN:
                raise PoleError()
        except (PoleError, ValueError):
            if l is not None:
                raise
            r = heuristics(e, z, z0, dir)
            if r is None:
                return self
```